src/dune_lang/decoder.ml

```diff
@@ -380,6 +382,47 @@ let pair a b = enter (a >>= fun a -> b >>= fun b -> return (a, b))
 let triple a b c =
   enter (a >>= fun a -> b >>= fun b -> c >>= fun c -> return (a, b, c))
 
+let unit_number name suffixes =
+  let unit_number_of_string ~loc s =
+    let l = String.length s in
+    let n, suffix =
+      let rec suffix_length i =
+        if i = l then
+          i
+        else
+          let c = s.[l - i - 1] in
+          if Char.code c >= Char.code '0' && Char.code c <= Char.code '9' then
+            i
+          else
+            suffix_length (i + 1)
+      in
+      String.split_n s (l - suffix_length 0)
+    in
+    let factor =
+      match List.assoc suffixes suffix with
+      | Some f -> f
+      | None ->
+        let possible_suffixes =
+          String.concat ~sep:", " (List.map ~f:fst suffixes)
+        in
+        User_error.raise ~loc
+          [ Pp.textf "invalid suffix, use one of %s" possible_suffixes ]
+    in
+    Option.map ~f:(( * ) factor) (Int.of_string n)
+  in
+  basic_loc name unit_number_of_string
+
+let duration = unit_number "Duration" [ ("s", 1); ("m", 60); ("h", 60 * 60) ]
+
+let bytes_unit =
+  unit_number "Byte amount"
+    [ ("B", 1)
+    ; ("kB", 1000)
+    ; ("KB", 1000)
+    ; ("MB", 1000 * 1000)
+    ; ("GB", 1000 * 1000 * 1000)
```

Comment:
Is it conventional to allow both `kB` and `KB` but not `mB` and `MB`? Just asking, I don't know the answer

Reply:
`kB`, `MB` and `GB` are the standardized units - e.g. a kilogram is `kg` and a ton is `Mg`. It felt strange to not accept `KB` since all other capitalized versions are accepted, however I never see `mB` and `gB`. I don't really mind, I did what felt sensible.

Not sure whether to get zealous with `kiB`, `MiB` etc either.

Reply:
Ok. Thanks for the details. The current state seems good to me.